### DISCUSSION TITLE
Run in development mode

### DISCUSSION THREAD

**Discussion by adwip** [coder/code-server#7058] [Q&A]
I tried to run code-server in development mode, so I can make little change in the source code, but after clone the repository and trying to run it as development mode by start typing this npm install, I got an error. Is there any way to do this?

**Comment by adwip** [comment]
Here the error

agung@usdd1:~/coder/nodejs/code-server$ npm install

> code-server@0.0.0 postinstall
> ./ci/dev/postinstall.sh

Installing dependencies for /home/agung/coder/nodejs/code-server/test
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'
npm error code 1
npm error path /home/agung/coder/nodejs/code-server
npm error command failed
npm error command sh -c ./ci/dev/postinstall.sh
npm error A complete log of this run can be found in: /home/agung/.npm/_logs/2024-10-30T17_53_01_765Z-debug-0.log

**Reply by code-asher** [reply]
Just to check, what version of Node and npm are you using?  Here is what I have:

```
λ node --version
v20.15.1
λ npm --version
10.7.0
```

What operating system is this?  I am not sure why the postinstall script is trying to get into a directory called `install`.  It should just be running `npm install`.  Could you edit `ci/dev/postinstall.sh` and add a `set -x` on the line after the other `set`, then run `npm install` again?  That should make it print every command it runs.

**Reply by adwip** [reply]
I'm running Ubuntu server on Raspberry pi 5. I have checked, the node version is 20.x.x (i forget the subversion x)

**Reply by code-asher** [reply]
Strange, that all seems like it should install just fine then.  Let me know if you are able to run the script with the debug flag described above.

**Reply by adwip** [reply]
Yes I will retry to run it again, maybe there is something wrong on my local raspi that prevent it to run the dev mode. I'll update here, thanks
